src/KlaviyoAPI.php

```diff
@@ -175,15 +175,15 @@ private function request( $method, $path, $options, $isPublic = false, $isV1 = f
     private function handleResponse( $response, $statusCode, $isPublic )
     {
         if ( $statusCode == 403 ) {
-            throw new KlaviyoAuthenticationException(self::ERROR_INVALID_API_KEY);
+            throw new KlaviyoAuthenticationException(self::ERROR_INVALID_API_KEY, $statusCode);
         } else if ( $statusCode == 404 ) {
-            throw new KlaviyoResourceNotFoundException(self::ERROR_RESOURCE_DOES_NOT_EXIST);
+            throw new KlaviyoResourceNotFoundException( self::ERROR_RESOURCE_DOES_NOT_EXIST, $statusCode);
         } else if ( $statusCode == 429 ) {
             throw new KlaviyoRateLimitException(
-                $this->returnRateLimit( $this->decodeJsonResponse( $response ) )
+                $this->returnRateLimit( $this->decodeJsonResponse( $response ), $statusCode )
             );
         } else if ( $statusCode != 200 ) {
-            throw new KlaviyoException( sprintf( self::ERROR_NON_200_STATUS, $statusCode ) );
+            throw new KlaviyoApiException($this->decodeJsonResponse( $response )['detail'], $statusCode);
```

Comment:
This should be totally fine without adjusting the message in the KlaviyoApiException constructor.